pandas/core/internals/managers.py

```diff
@@ -1705,26 +1706,42 @@ def construction_error(tot_items, block_shape, axes, e=None):
 def _form_blocks(arrays, names: Index, axes) -> List[Block]:
     # put "leftover" items in float bucket, where else?
     # generalize?
+
     items_dict: DefaultDict[str, List] = defaultdict(list)
     extra_locs = []
 
-    names_idx = names
-    if names_idx.equals(axes[0]):
-        names_indexer = np.arange(len(names_idx))
+    if len(arrays) == 1:
```

Comment:
for sure needs comments, and even better pls separate this out to a new function.

Reply:
Better?

Reply:
no, this looks almost like a groupby to me. i am puzzled why the axes are involved here at all.

Reply:
That's how it was already written. It finds all variables of the same block type, and makes blocks out of them. If you pass in more arrays than names, only those arrays whose names were specified are blockified.

I could rewrite this into a groupby, I guess. It'll create more contiguous blocks than it does currently.

If you look at `create_block_manager_from_blocks`, you'll see that it allows passing a singular array (not block), and it'll create a contiguous block for it. I could do this in `create_block_manager_from_arrays`, but IIRC that made a lot of things fail, like `init_dict`, which uses the same constructor. Maybe #38939 will fix that.